File: frontend/src/types/follow-up.ts
import type { InteractionChannel } from "@/types/interaction"

export const FOLLOW_UP_SOURCES = [
  "fixed_interval",
  "suggested_by_them",
  "manual",
] as const
export type FollowUpSource = (typeof FOLLOW_UP_SOURCES)[number]

export const FOLLOW_UP_SOURCE_LABELS: Record<FollowUpSource, string> = {
  fixed_interval: "Fixed interval",
  suggested_by_them: "Suggested by them",
  manual: "Manual",
}

export const FOLLOW_UP_STATUSES = [
  "pending",
  "done",
  "skipped",
  "cancelled",
] as const
export type FollowUpStatus = (typeof FOLLOW_UP_STATUSES)[number]

export const FOLLOW_UP_STATUS_LABELS: Record<FollowUpStatus, string> = {
  pending: "Pending",
  done: "Done",
  skipped: "Skipped",
  cancelled: "Cancelled",
}

export type FollowUp = {
  id: string
  threadId: string
  dueAt: string
  source: FollowUpSource
  intervalDays: number | null
  suggestedNote: string | null
  channelHint: InteractionChannel | null
  contactId: string | null
  status: FollowUpStatus
  completedAt: string | null
  completedInteractionId: string | null
  createdAt: string
  updatedAt: string
}

export type FollowUpInput = {
  threadId: string
  dueAt: string
  source: FollowUpSource
  intervalDays?: number | null
  suggestedNote?: string | null
  channelHint?: InteractionChannel | null
  contactId?: string | null
  status?: FollowUpStatus
  completedAt?: string | null
  completedInteractionId?: string | null
  remark?: string | null
}

File: backend/src/main/java/com/jat/jat/followup/domain/FollowUp.java
package com.jat.jat.followup.domain;

import com.jat.jat.interaction.domain.InteractionChannel;
import com.jat.jat.shared.domain.BaseAuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.time.Instant;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "follow_ups")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FollowUp extends BaseAuditableEntity {

    @Column(name = "thread_id", nullable = false)
    private UUID threadId;

    @Column(nullable = false)
    private Instant dueAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FollowUpSource source;

    private Integer intervalDays;

    private String suggestedNote;

    @Enumerated(EnumType.STRING)
    private InteractionChannel channelHint;

    @Column(name = "contact_id")
    private UUID contactId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private FollowUpStatus status = FollowUpStatus.pending;

    private Instant completedAt;

    @Column(name = "completed_interaction_id")
    private UUID completedInteractionId;

    @PrePersist
    void applyDefaults() {
        if (status == null) {
            status = FollowUpStatus.pending;
        }
        if (source == FollowUpSource.fixed_interval && intervalDays == null) {
            intervalDays = 2;
        }
    }
}
